packages/app/src/cli/utilities/extensions/name-mapper.ts

```diff
@@ -10,6 +10,8 @@ export function mapExtensionTypeToExternalExtensionType(extensionType: Extension
       return 'post_purchase_ui'
     case 'checkout_ui_extension':
       return 'checkout_ui'
+    case 'ui_extension':
+      return 'checkout_ui_beta'
```

Comment:
I've removed some duplication by calling `mapUIExternalExtensionTypeToUIExtensionType` from within `mapExternalExtensionTypeToExtensionType`, e.g:

```
export function mapExternalExtensionTypeToExtensionType(
  externalExtensionType: ExternalExtensionTypes | ExtensionTypes,
): ExtensionTypes {
  switch (externalExtensionType) {
    case 'post_purchase_ui':
    case 'checkout_ui':
    case 'checkout_ui_beta':
    case 'subscription_ui':
    case 'web_pixel':
    case 'pos_ui':
    case 'customer_accounts_ui':
      return mapUIExternalExtensionTypeToUIExtensionType(externalExtensionType)
    // etc
```

I did try to update `mapExternalExtensionTypeToExtensionType` so that typescript understood when it would return a UI extension vs an extension. My thinking here was that then you wouldn't even need `mapUIExternalExtensionTypeToUIExtensionType`, you could just use `mapExternalExtensionTypeToExtensionType`, but I didn't have any luck getting typescript to understand.  Would love any pointers if you have some.

These name-mapper functions are used to map the `--type` flag for the `yarn generate extension` command to the specification type that we use. E.g:

<img width="723" alt="Screen Shot 2022-11-02 at 10 16 26 AM" src="https://user-images.githubusercontent.com/690791/199513925-0a9f3fee-ecd7-423d-a9e3-ab3156b9eb3c.png">

I suppose that is because the options we output when you don't specify a flag aren't really ideal to be passed as a flag:

<img width="225" alt="Screen Shot 2022-11-02 at 10 22 10 AM" src="https://user-images.githubusercontent.com/690791/199514436-fa225c7b-1117-445d-a4a9-c0d7c5ad35b3.png">

Reply:
Thanks for the context!